packages/core/src/Popout/Popout.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { RefObject, useCallback, useEffect, useLayoutEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+import { Box } from '..'
+
+export interface IPopoutProps {
+  trigger: JSX.Element
+  content?: JSX.Element | undefined
```

Comment:
I normally use React.ReactElement as my type but I just randomly found that, is JSX.Element better?

Reply:
I use `JSX.Element` for just about the same reason you use `React.ReactElement`, I just randomly found it lol. From some skimming on StackOverflow, it looks like they are nearly equivalent, but using `JSX.Element` allows other JSX-ish libraries (like Preact, Solid, etc.) to still use the prop and understand what to do with it, even if they handle JSX their own way. 

Here's the post that I read through: [https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement](link)